app/src/main/java/com/github/libretube/repo/LocalFeedRepository.kt

```diff
@@ -70,6 +73,12 @@ class LocalFeedRepository : FeedRepository {
             }.filterNotNull().flatten().map(StreamItem::toFeedItem)
 
             DatabaseHolder.Database.feedDao().insertAll(collectedFeedItems)
+
+            // throttle feed extraction to avoid rate limiting
+            val count = extractionCount.getAndIncrement()
```

Comment:
We're fetching `CHUNK_SIZE` feeds in parallel above. Thus the counter would probably need to be incremented by `CHUNK_SIZE` instead of 1, right?

Or we move the counter incrementation to the try catch statement above, so that it's executed for each channel we fetch.